### DISCUSSION TITLE
IMAP IDLE support

### DISCUSSION THREAD

**Discussion by jav4** [Z-Hub/Z-Push#112] [Show and tell]
Hello all,

I'm testing an implementation of IMAP IDLE usage I've written for Z-Push. Looks fine, although I'm still checking and fine tuning. It reduces server polling to just a single IDLE connection, although it's limited to a single mailbox (it selects INBOX if given the choice).

It requires an IDLE implementation in the IMAP library so instead of the standard php-imap library I'm using https://github.com/javanile/php-imap2 with a custom-made idle function (I also needed OAUTH2).

Would that be of interest?

**Comment by kour1er** [comment]
@jav4 That sounds very interesting. I don't suppose it would be possible to share your code so I could experiment? Thanks!
